### PR TITLE
LPS-194204 | master

### PR DESCRIPTION
**Pull request by allen-nagtalon**
https://liferay.atlassian.net/browse/LPS-194204

### CODE REVIEW

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase [hunk 1]
```diff
@@ -2715,6 +2694,81 @@ definition {
 			searchAssetType = "Custom Object");
 	}
 
+	@priority = 3
```
Comment (brookedalton) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase:2697]: Hey! Just commenting for reference but can you add a description here? Thanks!

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase [hunk 2]
```diff
@@ -2715,6 +2694,81 @@ definition {
...
+			Navigator.openURL();
+
+			AssertElementNotPresent(locator1 = "Search#EMBEDDED_HEADER_SEARCH_BAR");
+
+			Click(locator1 = "Icon#CUSTOM_EDIT");
+
+			AssertVisible(locator1 = "Search#EMBEDDED_HEADER_SEARCH_BAR");
+		}
+
+		task ("When the user is signed out") {
```
Comment (brookedalton) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase:2717]: Combine with task below please

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase [hunk 3]
```diff
@@ -2715,6 +2694,81 @@ definition {
...
+			Click(locator1 = "Icon#CUSTOM_EDIT");
+
+			AssertElementNotPresent(locator1 = "Search#EMBEDDED_HEADER_SEARCH_BAR");
+
+			AssertVisible(
+				key_infoMessage = "This search bar is not visible to users yet.",
+				locator1 = "Message#INFO_SPECIFIC");
+		}
+
+		task ("When the user is signed out") {
```
Comment (brookedalton) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase:2761]: Combine this one also

**Comment by brookedalton** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase [hunk 4]
```diff
@@ -2694,6 +2694,78 @@ definition {
 			searchAssetType = "Custom Object");
 	}
 
+	@description = "This is a use case for LPS-193198."
+	@priority = 3
+	test ViewSearchBarVisibility {
+		task ("Given that the embedded search bar in the header is configured as invisible") {
```
Comment (joshchong) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase:2700]: no need to use the given, when, then verbiage for our tests. lean towards writing tasks like high level steps and assertions. For example, this one can be reworded to: `Configure the embedded header search bar to be invisible`

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase [hunk 5]
```diff
@@ -2694,6 +2694,78 @@ definition {
...
+			Search.openSearchPage();
+
+			SearchPortlets.configureEmbeddedSearchBar(invisible = "true");
+		}
+
+		task ("Then the search bar should be visible on the search/edit page, but not visible on the home page") {
```
Comment (joshchong) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase:2706]: there's a lot of assertions happening here that are significant enough to be separated into separate tasks. it's ok if a task just has a single assertion function in it.

example:
- Assert the search bar is visible on the search page
- Assert the search bar is not visible on the view mode of a content page
- Assert the search bar is visible within the edit mode of a content page

Comment (joshchong) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase:2706]: try to do this with the rest of the tasks in the test

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase [hunk 6]
```diff
@@ -2694,6 +2694,78 @@ definition {
...
+			Navigator.openURL();
+
+			AssertElementNotPresent(locator1 = "Search#EMBEDDED_HEADER_SEARCH_BAR");
+
+			Click(locator1 = "Icon#CUSTOM_EDIT");
+
+			AssertVisible(locator1 = "Search#EMBEDDED_HEADER_SEARCH_BAR");
+		}
+
+		task ("Given the user is signed out, then the search bar should not be present on the home page and the search page") {
```
Comment (joshchong) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase:2718]: another example:
- Sign out and assert the search bar is not visible on the view mode of a content page
- and that it is also not visible on the search page

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase [hunk 7]
```diff
@@ -2694,6 +2694,78 @@ definition {
...
+			AssertElementNotPresent(locator1 = "Search#EMBEDDED_HEADER_SEARCH_BAR");
+
+			AssertVisible(
+				key_infoMessage = "This search bar is not visible to users yet.",
+				locator1 = "Message#INFO_SPECIFIC");
+
+			Navigator.openURL();
+
+			AssertElementNotPresent(locator1 = "Search#EMBEDDED_HEADER_SEARCH_BAR");
+
```
Comment (joshchong) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase:2750]: can you also assert that the "This search bar is not visible..." message is not present as well. I forgot to put that in the ticket steps.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase [hunk 8]
```diff
@@ -2694,6 +2694,78 @@ definition {
...
+				key_infoMessage = "This search bar is not visible to users yet.",
+				locator1 = "Message#INFO_SPECIFIC");
+		}
+
+		task ("Given the user is signed out, then the search bar should not be visible on the search page") {
+			SignOut.signOut();
+
+			Search.openSearchPage();
+
+			AssertElementNotPresent(locator1 = "Search#EMBEDDED_HEADER_SEARCH_BAR");
```
Comment (joshchong) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase:2765]: assert that the "This search bar is not visible.." is not present as well

**Review by joshchong — CHANGES_REQUESTED** [review]
Thanks @allen-nagtalon! This is looking good for the most part besides some minor requested changes. let me know if you have any questions.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase (hunk 5 above)
Comment (brookedalton) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/Search.testcase:2706]: My bad on this - I asked for a lot of tasks to be grouped together 😬

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/search/Search.path [hunk 9]
```diff
@@ -90,6 +90,11 @@
 	<td>//div[contains(@id,'customRangeFrom')]//input</td>
 	<td></td>
 </tr>
+<tr>
+	<td>EMBEDDED_HEADER_SEARCH_BAR</td>
+	<td>//header[@id='banner']//input[contains(@data-qa-id,'searchInput')]</td>
+	<td></td>
+</tr>
```
[the remark this answers is not in the document]
Comment (allen-nagtalon) [reply portal-web/test/functional/com/liferay/portalweb/paths/portlet/search/Search.path:97]: I tested the following tests involving Search#EMBEDDED_SEARCH_BAR and got the following results:

- [PASSED] SearchBarSuggestions#ConfigureBlueprintContributorCharacterThresholds
- [PASSED] ClassicTheme#CanConfigureLookAndFeelSettingsOnSinglePage
- [PASSED] SitesWithPermissions#SitesTemplatePermissionPersistence
- [FAILED] CommerceSearch#SearchBarWidgetSearchProductsByCategory
- [FAILED] TeamExtranet#VerifySearchBarOnPages

**Comment by joshchong** [comment]
Looks good! Forwarded here with minor changes: https://github.com/brianchandotcom/liferay-portal/pull/140317

**Comment by joshchong** [comment]
ci:close

**Pull request closed without merging**